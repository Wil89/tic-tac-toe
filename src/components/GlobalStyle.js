import { createGlobalStyle } from "styled-components";

export const GlobalStyle = createGlobalStyle`
    body {
        font: 14px "Century Gothic", Futura, sans-serif;
        margin: 20px;
    }
`;