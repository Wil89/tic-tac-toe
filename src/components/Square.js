import styled from 'styled-components'
import React from 'react'

const defaultSquare = ({className, value, onClick}) => {
    return (
            <button data-testid="square" className={className} onClick={onClick}>{value}</button>
    )
}

export const Square = styled(defaultSquare)`
    background: #fff;
    border: 1px solid #999;
    float: left;
    font-size: 24px;
    font-weight: bold;
    line-height: 34px;
    height: 34px;
    margin-right: -1px;
    margin-top: -1px;
    padding: 0;
    text-align: center;
    width: 34px;
`;

