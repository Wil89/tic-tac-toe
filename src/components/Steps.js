import React from 'react'

export const Steps = ({move, onClick}) => {
    
    const desc = move ? `Go to move # ${move}` : 'Go to game start';  
    
    return (
        <li>
            <button data-testid="step" onClick={() => onClick(move)}>{desc}</button>
        </li>
            
    )
}
