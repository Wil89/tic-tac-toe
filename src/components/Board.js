import React, {useContext} from 'react'
import styled from 'styled-components'
import { HistoryContext } from '../context/Context';
import { Square } from './Square'

const BoardRow = styled.div`
    &:after{
        clear: both;
        content: "";
        display: table;
    }
`;

export const Board = () => {
    const {history, addMove, stepNumber} = useContext(HistoryContext)
    const current = history[stepNumber];
    
    const renderSquare = (i) => 
        <Square value={current.squares[i]} onClick={() => addMove(i)} />   

    return (
        <div>
            {/* <Status>{updateStatus()}</Status> */}
            <BoardRow>
                {renderSquare(0)}
                {renderSquare(1)}
                {renderSquare(2)}
            </BoardRow>
            <BoardRow>
                {renderSquare(3)}
                {renderSquare(4)}
                {renderSquare(5)}
            </BoardRow>
            <BoardRow>
                {renderSquare(6)}
                {renderSquare(7)}
                {renderSquare(8)}
            </BoardRow>
        </div>
    )
}
