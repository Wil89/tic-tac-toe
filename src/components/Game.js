import React, { useContext } from "react";
import styled from "styled-components";
import { HistoryContext } from "../context/Context";
import { winnerCheck } from "../utils/helper";
import { Board } from "./Board";
import { Steps } from "./Steps";

const GameContainer = styled.div`
  display: flex;
  flex-direction: row;
`;

const GameBoard = styled.div``;

const Status = styled.div`
  margin-bottom: 10px;
`;

const GameInfo = styled.div`
  margin-left: 20px;
`;

const StyledOL = styled.ol`
  padding-left: 30px;
`;

export const Game = () => {
  const { history, isXNext, jumpTo, stepNumber } = useContext(HistoryContext);

  const squares = history[stepNumber].squares;

  const getStatus = () => {
    //const squares = history[history.length - 1].squares;
    const winner = winnerCheck(squares);
    const nextPlayer = isXNext ? "X" : "O";
    return winner ? `Winner: ${winner}` : `Next Player: ${nextPlayer}`;
  };

  return (
    <div>
      <GameContainer>
        <GameBoard>
          <Board />
        </GameBoard>
        <GameInfo>
          <Status data-testid='status'>{getStatus()}</Status>
          <StyledOL>
            {history.map((step, move) => (
              <Steps key={move} move={move} onClick={jumpTo} />
            ))}
          </StyledOL>
        </GameInfo>
      </GameContainer>
    </div>
  );
};
