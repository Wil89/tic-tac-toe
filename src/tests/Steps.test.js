import { cleanup, fireEvent, render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Steps } from "../components/Steps";

describe("Board component tests", () => {
  beforeEach(cleanup);

  it("Steps match snapshot", () => {
    const { asFragment } = render(<Steps />);
    expect(asFragment(<Steps />)).toMatchSnapshot();
  });

  it("Describe initial value", () => {
    const { getByTestId } = render(<Steps />);
    expect(getByTestId("step")).toHaveTextContent("Go to game start");
  });

  it("Test event launched", () => {
    const onClick = jest.fn();
    const { getByTestId } = render(<Steps onClick={onClick} />);

    fireEvent.click(getByTestId("step"));
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
