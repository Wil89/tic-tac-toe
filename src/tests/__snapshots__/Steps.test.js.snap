// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board component tests Steps match snapshot 1`] = `
<DocumentFragment>
  <li>
    <button
      data-testid="step"
    >
      Go to game start
    </button>
  </li>
</DocumentFragment>
`;
